lasagne/layers/noise.py

```diff
@@ -211,3 +219,76 @@ def get_output_for(self, input, deterministic=False, **kwargs):
             return input + self._srng.normal(input.shape,
                                              avg=0.0,
                                              std=self.sigma)
+
+
+class AlphaDropoutLayer(DropoutLayer):
+    """Dropout layer.
+
+    Sets values to alpha. Will also converge the remaining
+    neurons to mean 0 and a variance of 1 unit. See notes for disabling dropout
```

Comment:
Sorry, this explanation is a bit imprecise. Can you be more explicit what the layer does? (If you can't find anything, I'll try that myself.)